mlflow/cli.py

```diff
@@ -136,16 +137,23 @@ def run(uri, entry_point, version, param_list, docker_args, experiment_name, exp
         sys.exit(1)
 
 
-def _user_args_to_dict(user_list, flag_name='P'):
+def _user_args_to_dict(arguments, argument_type='P'):
     user_dict = {}
-    for s in user_list:
-        index = s.find("=")
-        if index == -1:
+    for arg in arguments:
+        split = arg.split('=')
+        # Docker arguments such as `t` don't require a value
+        # -> set them to True if specified
+        if len(split) == 1:
+            if argument_type == 'A':
+                name = split[0]
+                value = True
```

Comment:
Fine with you if I work with booleans here or do you want me to use 'true'?

Reply:
@Zethson How do we intend to map True (or `true`) to the proper command flag (which is just `-t` rather than `-t True` or `-t true`)?

Reply:
Perhaps the user args to do not necessarily have to be a dictionary, since certain args do not have value elements. We may consider changing the return type of this to a list of `UserArgument` objects with two properties: `name` (required) and `value` (optional).

Reply:
Not sure whether I understand the first question.
If only a single parameter is specified without `=value`, it is automatically set to the value True. This only happens when the parameter is passed. All command line interfaces treat options like this. Note that this behavior is only enabled for Docker arguments and not for mlflow -P parameters. 

About the second question: If mlflow were at Python 3.7 we could use a pretty [data class](https://docs.python.org/3/library/dataclasses.html). However, since this is not the case I guess it would not make the code much prettier. 

>  since certain args do not have value elements. 

Well, yes and no. They don't have an associated value when passed by the user, but they automatically get the value True inside mlflow. At runtime every argument has a name AND a value.

I am fine with the current design, but if you have more specific wishes (and reasoning) I am of course happy to adjust the implementation.

Reply:
> If only a single parameter is specified without `=value`, it is automatically set to the value True. 

I think the issue is that this behavior is incompatible with downstream CLI tools like Docker. For example, if `_user_args_to_dict` returns a dictionary with a string -> boolean mapping which is then passed to `_get_docker_command`, the resulting Docker command will contain an argument of the form `--name True`, which is not the correct behavior for boolean flags on the commandline.

`docker run --interactive True ...` is *invalid*
`docker run --interactive ...` is valid.

I think defining `user_args_to_dict` like this is fine, but the downstream consumer of this dictionary (i.e., `_get_docker_command`) needs to handle boolean values in the dictionary. Specifically, the string `True` or `False` should not be passed to the `docker run` command. Let me know if this makes sense.

Reply:
Yeah, I'm aware of that. Hence, the manual setting of the -t in the command.

The only thing that makes this more complicated is that all the options that may contain boolean values need to be passed BEFORE the rest of the a=b arguments.

Hence, I suggest that I write some code that filters all pairs of name:booleanvalue -> set them first by appending `-name` to the docker run command -> removing them from the dict (or use a prefiltered one) and then append the rest as is business now.
Since the dict is always super small there is no performance penalty anyways.

Does this sound fine to you?

Reply:
This would of course also allow for all boolean options (not just -t like now).
Though I fail to see why you would ever pass -i  :)

Reply:
Hmmm...I'm probably missing something, but I don't understand why boolean arguments need to come first. Can you clarify?

Reply:
Huh, actually I thought so, but this doesn't seem to be the case.
Well yeah, then things get easier of course.

Reply:
I need to sleep, but will get back to you with a new, certainly cleaner looking draft tomorrow.

Reply:
Awesome!